File: solutions/elixir/047-rotate-list.ex
# Problem 47: Rotate List (LeetCode 61)
# Difficulty: Med
# Language: Elixir
# 
defmodule RotateList do
  def rotate_right([], _k), do: []
  def rotate_right(list, k) do
    n = length(list)
    rot = rem(k, n)
    if rot == 0, do: list,
    else: Enum.drop(list, n - rot) ++ Enum.take(list, n - rot)
  end
end

File: solutions/typescript/047-rotate-list.ts
/**
 * Problem 47: Rotate List (LeetCode 61)
 * Difficulty: Med
 * Language: TypeScript
 */
function rotateRight(head: number[], k: number): number[] {
  if (head.length === 0) return [];
  const n = head.length;
  const rot = k % n;
  if (rot === 0) return head;
  return [...head.slice(n - rot), ...head.slice(0, n - rot)];
}
